Dockerfile.e2e-tests

```diff
@@ -1,5 +1,5 @@
 # Taking a non-alpine image for e2e tests so that cgo can be enabled for the race detector.
-FROM golang:1.23.3 as builder
```

Comment:
Nit/question: this was using a patch tag before, while the reference below was using a minor. Should we use 1.24.0 here?

Reply:
:+1: let's use 1.24.0 so that it wouldn't change accidentally